pkg/clients/ec2/instance.go

```diff
@@ -503,6 +503,20 @@ func GenerateEC2InstanceIPV6Addresses(addrs []manualv1alpha1.InstanceIPv6Address
 	return nil
 }
 
+// GenerateEC2Ipv6PrefixSpecificationRequest coverts an internal slice of Ipv6PrefixSpecificationRequest into a slice of ec2.Ipv6PrefixSpecificationRequest
+func GenerateEC2Ipv6PrefixSpecificationRequest(prefixes []manualv1alpha1.Ipv6PrefixSpecificationRequest) []types.Ipv6PrefixSpecificationRequest {
+	if prefixes != nil {
```

Comment:
```suggestion
	if len(prefixes) == 0 {
	  return nil
	}
```
Returning early would be easier to follow and checking for array length would be more robust to prevent cases where we send empty array to AWS which get rejected in some cases.